web/src/lib/components/SelectLocation.svelte

```diff
@@ -0,0 +1,243 @@
+<script lang="ts">
+	import { onDestroy, onMount } from "svelte";
+	import OlMap from "ol/Map";
+	import Modal from "../clients/Modal.svelte";
+	import Map from "./map/Map.svelte";
+	import VectorLayer from "ol/layer/Vector";
+	import { Feature, MapBrowserEvent } from "ol";
+	import { Point } from "ol/geom";
+	import VectorSource from "ol/source/Vector";
+	import Button from "./Button.svelte";
+	import type { Coordinate } from "ol/coordinate";
+	import {
+		convertToMapProjection,
+		convertToResourceProjection,
+	} from "../utils/map";
+	import { t } from "../utils/i8n";
+	import ecomapHttpClient from "../clients/ecomap/http";
+	import { getLocationName } from "../utils/location";
+
+	/**
+	 * Indicates if the modal is open.
+	 */
+	export let open: boolean;
+
+	/**
+	 * Callback fired when open state of the modal changes.
+	 * @param open New open state modal.
+	 */
+	export let onOpenChange: (open: boolean) => void;
+
+	/**
+	 * Callback fired when save action is triggered.
+	 * @param coordinate Container coordinate.
+	 * @param locationName Container location name.
+	 */
+	export let onSave: (coordinate: Coordinate, locationName: string) => void;
+
+	/**
+	 * Coordinate to display in the map.
+	 * @default null
+	 */
+	export let coordinate: Coordinate | null = null;
+
+	/**
+	 * Callback fired when cancel action is triggered.
+	 * @default null
+	 */
+	export let onCancel: (() => void) | null = null;
+
+	/**
+	 * Open Layers map.
+	 */
+	let map: OlMap;
+
+	/**
+	 * ID of the selected location.
+	 */
+	const featureId = "location";
+
+	/**
+	 * The map layer which displays the selected location.
+	 */
+	const layer = new VectorLayer({
+		source: new VectorSource<Feature<Point>>({ features: [] }),
+		style: {
+			"icon-src": "/images/pin.svg",
+		},
+	});
+
+	/**
+	 * Indicates if save action is disabled.
+	 * @default true
+	 */
+	let disabled = true;
```

Comment:
The documentation is good, but the name of this variable is too generic in my opinion